buddies-server/routes/users.js

```diff
@@ -1,99 +1,94 @@
         const salt = await bcrypt.genSalt(10);
         req.body.password = await bcrypt.hash(req.body.password, salt);
-      } catch (err) {
-        return res.status(500).json(err);
       }
-    }
-    try {
-      const user = await User.findByIdAndUpdate(req.params.id, {
+
+      await User.findByIdAndUpdate(req.params.id, {
         $set: req.body,
       });
-      res.status(200).json("Account has been updated");
-    } catch (err) {
-      return res.status(500).json(err);
+
+      return res.status(200).json("Account has been updated");
+    } else {
+      return res.status(403).json("You can update only your account!");
     }
-  } else {
-    return res.status(403).json("You can update only your account!");
-  }
-});
+  })
+);
 
 //delete user
-router.delete("/:id", async (req, res) => {
-  if (req.body.userId === req.params.id || req.body.isAdmin) {
-    try {
+router.delete("/:id", 
+  requiresRole("user"),
+  asyncHandler(async (req, res) => {
+    if (req.user.id === req.params.id || req.user.permissions.includes("admin")) {
       await User.findByIdAndDelete(req.params.id);
-      res.status(200).json("Account has been deleted");
-    } catch (err) {
-      return res.status(500).json(err);
+      return res.status(200).json("Account has been deleted");
+    } else {
+      return res.status(403).json("You can delete only your account!");
     }
-  } else {
-    return res.status(403).json("You can delete only your account!");
-  }
-});
+  })
+);
 
 //get a user
-router.get("/:id", async (req, res) => {
-  try {
-    const user = await User.findById(req.params.id);
-    const { password, updatedAt, ...other } = user._doc;
-    res.status(200).json(other);
-  } catch (err) {
-    res.status(500).json(err);
-  }
-});
+router.get("/:id", asyncHandler(async (req, res) => {
+  const user = await User.findById(req.params.id);
+  const { password, updatedAt, ...other } = user._doc;
 
-//follow a user
+  return res.status(200).json(other);
+}));
 
-router.put("/:id/follow", async (req, res) => {
-  if (req.body.userId !== req.params.id) {
-    try {
+// follow a user
+router.put("/:id/follow", 
+  requiresRole("user"),
+  asyncHandler(async (req, res) => {
+    if (req.user.id !== req.params.id) {
       const user = await User.findById(req.params.id);
       const currentUser = await User.findById(req.body.userId);
+
       if (!user.followers.includes(req.body.userId)) {
         await user.updateOne({ $push: { followers: req.body.userId } });
         await currentUser.updateOne({ $push: { followings: req.params.id } });
-        res.status(200).json("user has been followed");
+        return res.status(200).json("user has been followed");
       } else {
-        res.status(403).json("you allready follow this user");
+        return res.status(403).json("you already follow this user");
       }
-    } catch (err) {
-      res.status(500).json(err);
+    } else {
+      returnres.status(403).json("you cant follow yourself");
```

Comment:
⛔ There is a typo here! `returnres` should be `return res`

Reply:
Oops. My bad

Reply:
Pushed a new commit to resolve this